dev/tools/localization/gen_l10n.dart

```diff
@@ -913,8 +940,25 @@ class LocalizationsGenerator {
     // Generate the required files for localizations.
     _languageFileMap.forEach((File file, String contents) {
       file.writeAsStringSync(contents);
+      if (_inputsAndOutputsListFile != null) {
+        _outputFileList.add(file.absolute.path);
+      }
     });
-    outputFile.writeAsStringSync(_generatedLocalizationsFile);
+
+    baseOutputFile.writeAsStringSync(_generatedLocalizationsFile);
+    if (_inputsAndOutputsListFile != null) {
+      _outputFileList.add(baseOutputFile.absolute.path);
+
+      // Generate a JSON file containing the inputs and outputs of the gen_l10n script.
+      if (!_inputsAndOutputsListFile.existsSync()) {
+        _inputsAndOutputsListFile.createSync(recursive: true);
+      }
+      _inputsAndOutputsListFile.writeAsStringSync(
+        inputAndOutputsJsonFileTemplate
+          .replaceAll('@(inputs)', _inputFileList.map((String file) => '"$file"').join(',\n    '))
```

Comment:
Instead of using string replace, create a map and use json.encode:

```
_inputsAndOutputsListFile.writeAsStringSync(
  json.encode(<String, Object>{
    'inputs': _inputFileList,
    'outputs': _outputFileList,
  }),
)
```